README.md

```diff
@@ -82,9 +82,12 @@ Should start with `is`, `has`, `will` or `should`. Like `isValid` or `hasValues`
 #### Always Affirmative
 Avoid negations. _“Don’t ever not avoid negative logic”_. Prefer `isShown` over `isHidden` or `isEnabled` over `isDisabled`. Do not use names like `notEditable`.
 
+#### TypeScript Generics
+If a Class or Interfaces uses only one generic and the usages of the generic is obvious, then naming that generic `T` is sufficient. If a Class or Interface has more generics or when the usage is not obvious, you should use a more descriptive name. The same naming rules as for classes will apply then. 
```

Comment:
Even if more than 1, you can continue `T`, with `U`, `V`, etc, as long as it's clear enough.

Reply:
I started out on a library with many generic parameters and writing descriptive param names which became very, very verbose rather quickly. It also introduced name collisions fast because if you want to extend a generic param type, it makes sense to want to then use the same name, i.e.; `<SomeType extends SomeType>`. 

What I’ve then found is when param ambiguity becomes an issue, it's likely because its type is not so generic after all, and you should have it extend another type, this has the added benefit of being more type safe, and it labels the param, making it more clear what it represents, i.e.: `<T extends SomeType>`, now it is clear `T` represents `SomeType`.